### ISSUE TITLE
Identify initial set of trees to populate into treetracker.tree

### ISSUE THREAD

**Issue by ZavenArra** [Greenstand/treetracker-api#47]
Identify initial freetown captures and/or Identify initial forestmatic captures

Create tree records for these and link the capture record.

**Comment by ZavenArra** [comment]
Initial set of trees should be the freetown captures with note = 1, and then we should load ALL freetown captures into capture table.

**Comment by ZavenArra** [comment]
select note from trees where planting_organization_id IN (select entity_id from getEntityRelationshipChildren(178));

**Comment by ZavenArra** [comment]
select note from trees join planter on trees.planter_id = planter.id where planter.organization_id IN (select entity_id from getEntityRelationshipChildren(178));

**Comment by sebastiangaertner** [comment]
Yes this is the way to identify the first capture of a tree from Freetown only. (178 and its sub orgs). This is still true and should be the base of the tree entries under Freetown.

**Comment by sebastiangaertner** [comment]
```
SELECT *
FROM public.trees t
join planter p on t.planter_id = p.id
WHERE p.organization_id in (select entity_id from getEntityRelationshipChildren(178))
and length(t.note) > 4;
```

**Comment by camwebb** [comment]
I'm looking into the potential to use the Freetown paper tag information in the notes field to generate a test set of captures and known recaptures. I queried the DB as above adding a tag regexp of `t.note ~ '^ *[A-Za-z]+[0-9]+ *[a-z]+ *[0-9]+ *$'` and got 234,746 captures. Yay! Then I wrote a [script](https://gist.github.com/camwebb/44b37d7f850ee5835c4396c71c172824) to find multiple instances of the same tag, and generate a simple webpage so we can view them. 

Some results:

Distribution of number of instances of the same tag (mostly 1, but up to 28):
![freetown](https://user-images.githubusercontent.com/1167435/201435762-61abbdfb-20b6-46f9-af05-2017dadafcbb.jpg)

The maximum date range between captures having the same tag was 341 days, but most duplicate occurrences of the tag were on the same date - i.e., the notes field was (auto?) copied into a subsequent capture of a different plant.

You can see the capture matching pages here:

 * 100 cases of four capture instances of the same tag: [view](https://camwebb.info/tmp/freetown_tag_test_1.html)
 * 100 cases of two capture instances of the same tag with a minimum of 50 days between captures: [view](https://camwebb.info/tmp/freetown_tag_test_2.html)

The conclusion: it appears we cannot use the tag information in the notes field. Except for a few rare cases, the plants are clearly not the same ones 🙁  This does not mean that there are no true recaptures, but just that repeated tag codes in the notes field appears to be too often erroneous to use.

(One possible modification to the filter in the script is to limit the candidates for true recaptured trees to those within some number of metres apart, using the GPS locations. No time now though.)

**Comment by dadiorchen** [comment]
@camwebb the SQL above is to generate tree, that means, the first capture, if I understand this correctly, all those captures matching this condition are different trees, so we shouldn't use the same note to bind trees, right @sebastiangaertner ?

**Comment by dadiorchen** [comment]
@Kpoke I think we should use treetracker.capture as the source of this operation, right? Migrate treetracker.capture first, then do this?

**Comment by Kpoke** [comment]
Oh yeah definitely that's the only way to go, trees->raw-captures->captures->trees

**Comment by camwebb** [comment]
@dadiorchen Unless I am not understanding something major, each entry in the `trees` table is just a photograph (=capture) and so two or more records in that table can be of the same physical plant (see my [previous experiment](https://camwebb.info/files/webb2022_thtp_recaptures.pdf)). If a tree is revisited, and the tag number is entered, it will appear in two records. Therefore we _should_ be trying to use the tag to bind captures of the same tree together. But my study here indicates that the tag codes in the notes field cannot be trusted.

**Comment by camwebb** [comment]
@Kpoke @dadiorchen Surely the best way to create a small test set for matching is to get external information from Freetown planting orgs on which physical locations have be subject to repeated visits, during which existing seedlings were re-photographed? Then just filter `trees` for the appropriate long/lat bounding box This. is effectively what I did in my previous [study](https://camwebb.info/files/webb2022_thtp_recaptures.pdf). Do we have any such information that can help pick a test location?

**Comment by dadiorchen** [comment]
>@dadiorchen Unless I am not understanding something major, each entry in the trees table is just a photograph (i.e., = "capture", not "tree") and so two or more records in that table can be of the same physical plant (see my [previous experiment](https://camwebb.info/files/webb2022_thtp_recaptures.pdf)). If a tree is revisited, and the tag number is entered, it will appear in two records. Therefore we should be trying to use the tag to bind captures of the same tree together. But my study here indicates that the tag codes in the notes field cannot be trusted.

Here tag means the `note` field in the tree table, right? But in my understanding, Freetown guys use that note as a batch id to tag a bundle of trees when they planting them, that's why captures with the same tags were taken at almost the same date.
Then, the mathching tool will be used to bind the following capture indicating the same tree, we can do it using the GPS location.

**Comment by dadiorchen** [comment]
>@Kpoke @dadiorchen Surely the best way to create a small test set for matching is to get external information from Freetown planting orgs on which physical locations have be subject to repeated visits, during which existing seedlings were re-photographed? Then just filter trees for the appropriate long/lat bounding box This is effectively what I did in my previous [study](https://camwebb.info/files/webb2022_thtp_recaptures.pdf). Do we have any such information that can help pick a test location?

Yes, I agree, we can use a specified area for testing, like pick a shape on the may by GIS tool

**Comment by sebastiangaertner** [comment]
@camwebb the methodology on the ground executed by the trackers was the following: on first visit of the tree after planting the notes fields should have a species name in it. Some tracker managed to follow through the complete coding which was  for example "KlC12tombi851" = Catchment/Species/physicallytaggedID  
Any retracking was supposed to only have a number in it such as 1 for the first 2 for the second ... etc. 
Some trackers have obviously put the year in so we have entries like 2022 or even 20222022.  It is the typical garbage in garbage out scenario that we are dealing with. 
We do have additional time ranges to get a better quality. 
So if you create a filter as you did and it includes the characters within a note field that would suggest its a tree being registered. 
If its only numbers it can be sure to be a re-visit.
I think a script that identifies the obvious ones (based on the rule above) + looking at the "session" of that planter ie. `planter_id `and `time_created` assuming that that session has consistent quality there. This would even out human error which surely appears to be the case.

**Comment by camwebb** [comment]
@sebastiangaertner 

> Some tracker managed to follow through the complete coding which was for example "KlC12tombi851" = Catchment/Species/physicallytaggedID
> Any retracking was supposed to only have a number in it such as 1 for the first 2 for the second ..

So, to be clear, their protocol does not allow us to link recaptures by tag? - dang that's a huge lost opportunity.

So I guess the sampling protocol is:

 1. Locate a smallish spatial area (1 square km?) that contains a sufficient number of retags (`t.note ~ '^ *[0-9] *$'`).
 2. Query this long/lat bounding box for _all_ captures, somehow flagging the recaptures (R) and non-recaptures (=originals, O).
 3. Populated the DB tables with this test set of captures.
 4. "for each R..." run the capture matching tools, which I assume by now filter potential candidates based on GPS distance, to find an "O".

For step 1, a test site:

```
SELECT replace(t.note,' ','') AS recapture, count(*) FROM public.trees t JOIN planter p ON t.planter_id = p.id
WHERE p.organization_id in (select entity_id from getEntityRelationshipChildren(178))
AND t.lon > -13.25 AND t.lon < -13.24 AND t.lat > 8.44 AND t.lat < 8.45 
AND t.note ~ '^ *[0-9] *$' group by replace(t.note,' ','');

 reacpture | count 
-----------+-------
 1         |  1163
 2         |   325
 3         |  1289

SELECT count(*) FROM public.trees t JOIN planter p ON t.planter_id = p.id
WHERE p.organization_id in (select entity_id from getEntityRelationshipChildren(178))
AND t.lon > -13.25 AND t.lon < -13.24
AND t.lat > 8.44 AND t.lat < 8.45;

 count 
-------
  7110
```
I.e., ~7000 captures total, ~3000 marked as recaptures, so about ~4000 seedlings. Something like this might work.
